libr/core/canal.c

```diff
@@ -1926,12 +1926,21 @@ R_API int r_core_anal_esil_fcn(RCore *core, ut64 at, ut64 from, int reftype, int
 	return 0;
 }
 
+static int find_sym_flag(RAnalFunction *fcn, RFlagItem *f) {
+	return f->space && !strcmp (f->space->name, R_FLAGS_FS_SYMBOLS)? 0: 1;
+}
+
+static bool is_invalid_addr(RCore *core, ut64 addr) {
```

Comment:
just this function just call r_io_is_valid api to determine if its mapped or not? if theres a function or a flag doesnt matters much imho. also i think this slowdowns the analysis

Reply:
IDK who wrote this check and why it prints error "Invalid address...". I just extended it, moved it to a function and named it `is_invalid_addr`.

Reply:
I can rename it to avoid confusions. Maybe `skip_addr`?

Reply:
I don't see performance loss on the results of performed tests. The check is involved only when `r_core_anal_fcn()` is called inside `aa`, `af` and `afu`.

Reply:
what this function is doing is not checking if the address is valid. this is what RIO.is_valid_addr() do.

In this case this is chcking if there's a function or a flag in there. Fix the function name to clarify this and LGTM